File: android/app/src/main/java/com/hogent/android/database/entities/Customer.kt
package com.hogent.android.database.entities

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass


data class Customer(
    @Json(name = "lastname")
    val lastName: String,
    @Json(name = "firstname")
    val firstName: String,
    @Json(name = "phonenumber")
    val phoneNumber: String,
    val email: String,
    val password: String,
    var bedrijf_opleiding: String = Course.NONE.toString(),
    var contactPs1: ContactDetails1? = ContactDetails1(),
    var contactPs2: ContactDetails2? = ContactDetails2(),
    val id: Int = 0
)
@JsonClass(generateAdapter = true)
data class ContactDetails1(
    var contact1_phone: String? = "",
    var contact1_email: String? = "",
    var contact1_firstname : String? = "",
    var contact1_lastname: String? = ""
)
@JsonClass(generateAdapter = true)
data class ContactDetails2(
    var contact2_phone: String? = "",
    var contact2_email: String? = "",
    var contact2_firstname : String? = "",
    var contact2_lastname: String? = ""
)


enum class Course{
    NONE,
    TOEGEPASTE_INFORMATICA,
    AGRO_EN_BIOTECHNOLOGIE,
    BIOMEDISCHE_LABORATORIUMTECHNOLOGIE,
    CHEMIE,
    DIGITAL_DESIGN_AND_DEVELOPMENT,
    ELEKTROMECHANICA,
}

File: backend/src/repository/customer.js
const {tables, getKnex} = require("../data/index");

const getAll = async () => {
  return await getKnex()(tables.customer).select();
}

const updateCustomerById = async (id, customer) => {
    await getKnex()(tables.customer)
          .update({
            "contact1_phone" : customer.contactPs1.contact1_phone,
            "contact1_email" : customer.contactPs1.contact1_email,
            "contact1_firstname": customer.contactPs1.contact1_firstname,
            "contact1_lastname": customer.contactPs1.contact1_lastname,
            "contact2_phone" : customer.contactPs2.contact2_phone,
            "contact2_email" : customer.contactPs2.contact2_email,
            "contact2_firstname": customer.contactPs2.contact2_firstname,
            "contact2_lastname": customer.contactPs2.contact2_lastname,
          })
          .where("id", id)
    return id
}

const getCustomerById = async (id) => {
  return await getKnex()(tables.customer).select().where("id",id)
}

const registerCustomer = async (customer) => {
  const [id] = await getKnex()(tables.customer).insert({
    firstname: customer.firstname,
    lastname : customer.lastname ,
    email : customer.email,  
    password : customer.password,
    phonenumber: customer.phonenumber,
    bedrijf_opleiding : customer.bedrijf_opleiding,

    contact1_phone : customer.contactPs1.contact1_phone,
    contact1_email : customer.contactPs1.contact1_email,
    contact1_firstname : customer.contactPs1.contact1_firstname,
    contact1_lastname : customer.contactPs1.contact1_lastname,

    contact2_phone : customer.contactPs2.contact2_phone,
    contact2_email : customer.contactPs2.contact2_email,
    contact2_firstname : customer.contactPs2.contact2_firstname,
    contact2_lastname : customer.contactPs2.contact2_lastname,
  });
    return id
}

const loginCustomer = async (email,password) => {
  const customer =  await getKnex()(tables.customer)
    .where("email", email)
    .where("password", password)
  return customer
}

module.exports = {
  getAll,
  getCustomerById,
  updateCustomerById,
  registerCustomer,
  loginCustomer
}